### DISCUSSION TITLE
save log in checkpoint

### DISCUSSION THREAD

**Discussion by kingjr** [Lightning-AI/pytorch-lightning#7703] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi

Is there a way to save and retrieve data from the logger when we save/load a training checkpoint e.g.

```python
class CustomLogger(LightningLoggerBase):
  def __init__(self,):
      self.history = list()

  ...

  def log_metrics(self, metrics, step):
        self.history.append(metrics)

   def on_save_checkpoint(self, checkpoint):
       checkpoint['logger_history'] = self.history
   def on_load_checkpoint(self, checkpoint):
       self.history = checkpoint['logger_history']
```

thank you !

**Comment by sjgosai** [comment]
Hi,

I also have a use case where I would like to access the logs for recorded scalar metrics in my script after training but before saving my model. I've tried fiddling with `trainer.logger.experiment` but I can't locate a specific attribute that houses a metric dictionary. Alternatively, I'm using the `ModelCheckpoint` callback, and if I can the logged metrics for just the top_k saved, that would suffice.

Thanks!
